base/src/main/java/com/fasterxml/jackson/jakarta/rs/cfg/AnnotationBundleKey.java

```diff
@@ -129,13 +129,22 @@ private final boolean _equals(Annotation[] otherAnn)
         
         switch (len) {
         default:
-            for (int i = 0; i < len; ++i) {
-                if (!_annotations[i].equals(otherAnn[i])) {
+            for (Annotation annotation1 : _annotations) {
+                boolean found = false;
+                for (Annotation annotation2 : otherAnn) {
+                    if (annotation1 == annotation2) {
+                        found = true;
+                        if (!annotation1.equals(annotation2)) {
```

Comment:
Since you are already checking(in line 135) if both objects(annotation1 and annotation2) refer to the same location, is it necessary to have this condition here? Does this become true at all?

Reply:
Thanks @prathyushreddylpr for the feedback. Nice catch. I made the changes, Can you review it again?

Reply:
Looks good to me

Reply:
After the changes @prathyushreddylpr mentioned the code looks good to me.